src/server/routes/api/document/export.js

```diff
@@ -99,4 +109,105 @@ const exportToZip = (baseUrl, zipPath, types, biopaxIdMapping) => {
   return processNext();
 };
 
-export { exportToZip, EXPORT_TYPES };
+/**
+ * scheduleTask
+ * Schedule a task in 'delay' ms. Ignore additional requests while scheduled.
+ *
+ * @param {object} task The task to execute
+ * @param {number} delay ms delay for task execution (default 0)
+ * @param {object} next The callback to run after a task is initiated
+ */
+const taskScheduler = ( task, delay = 0, next = () => {} ) => {
+  let taskScheduled = false;
+  let taskTime = null;
+  const resetTaskSchedule = () => { taskScheduled = false; taskTime = null; };
+
+  return (() => {
+    const setTimeoutPromise = util.promisify( setTimeout );
+    let now = new Date();
+    logger.info( `A task request has been received` );
+
+    if( taskScheduled ){
+      logger.info( `A task has already been scheduled for ${taskTime} (${formatDistance( now, taskTime )})` );
+
+    } else {
+      taskTime = addMilliseconds( new Date(), delay );
+      logger.info( `A task was scheduled for ${taskTime} (${formatDistance( now, taskTime )})` );
+      taskScheduled = true;
+
+      setTimeoutPromise( delay )
+        .then( task )
+        .then( next )
+        .catch( () => {} ) // swallow
+        .finally( resetTaskSchedule ); // allow another backup request
+    }
+
+    return Promise.resolve();
+  });
+};
+
+// Configure Changefeeds for the document table
+const setupChangefeeds = async ({ rethink: r, conn, table }) => {
+  const docOpts = {
+    includeTypes: true,
+    squash: true
+   };
+
+  // Database restore of doc with public status
+  const toPublicStatusFromNull = r.row( 'new_val' )( 'status' ).eq( 'public' )
+    .and( r.row( 'old_val' ).eq( null ) );
+  // Status changed to 'public'
+  const toPublicStatusFromOtherStatus = r.row( 'new_val' )( 'status' ).eq( 'public' )
+    .and( r.row( 'old_val' )( 'status' ).ne( 'public' ) );
+  // Status is changed from 'public'
+  const toOtherStatusFromPublicStatus = r.row( 'new_val' )( 'status' ).ne( 'public' )
+    .and( r.row( 'old_val' )( 'status' ).eq( 'public' ) );
+
+  const docFilter = toPublicStatusFromNull.or( toPublicStatusFromOtherStatus ).or( toOtherStatusFromPublicStatus );
+  const cursor = await table.changes( docOpts )
+   .filter( docFilter )
+   .run( conn );
+  return cursor;
+};
+
+/**
+* initExportTasks
+* Initialize the export tasks
+*/
+const initExportTasks = async ( exportOnInit = true ) => {
+  const MS_PER_SEC = 1000;
+  const SEC_PER_MIN = 60;
+  const MIN_PER_HOUR = 60;
+  const baseUrl = `http://localhost:${PORT}`; // ever not localhost?
```

Comment:
Not really needed anyway, since you could call the corresponding function directly instead of going through the route.  It works either way, though going the routes could slow things down.

Reply:
Makes sense - this requires extracting the logic from the route handlers:
  - `/api/document/` and also
  - `/api/document/biopax`, 
  - `/api/document/sbgn`
  - `/api/document/text`

I'll try, but if it gets hairy I'll just leave it.

Reply:
- Refactoring converter functions
  - Stripped out the logic from the endpoints and reused them directly, avoiding the web service routes.
- Fixed up the behaviour on boot: 
  - When `NODE_ENV` is `production`, always create exports
  - When `NODE_ENV` is not `production` (e.g. dev), create the export zips if any are missing (so the web service routes work), otherwise do nothing